backend-lib/src/main/rust/lib.rs

```diff
@@ -3417,6 +3418,31 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_model_TorWalletClient_che
     unwrap_exc_or(&mut env, res, ptr::null_mut())
 }
 
+fn encode_address_check_result<'a, P: Parameters>(
+    env: &mut JNIEnv<'a>,
+    network: &P,
+    found: Option<TransparentAddress>,
+) -> jni::errors::Result<JObject<'a>> {
+    match found {
+        None => {
+            let nf_class = env.find_class(
+                "cash/z/ecc/android/sdk/internal/model/JniAddressCheckResult$NotFound",
+            )?;
+
+            let instance_sig =
+                "Lcash/z/ecc/android/sdk/internal/model/JniAddressCheckResult$NotFound;";
+
+            let value = env.get_static_field(nf_class, "INSTANCE", instance_sig)?;
+            value.l()
+        }
+        Some(address) => env.new_object(
+            "cash/z/ecc/android/sdk/internal/model/JniAddressCheckResult$Found",
+            "(java/lang/String;)V",
```

Comment:
This is the line that @Milan-Cerovsky noted is causing an error: https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1832#pullrequestreview-3418397455